gax-java/gax/src/main/java/com/google/api/gax/rpc/TransportChannelProvider.java

```diff
@@ -90,13 +90,19 @@ public interface TransportChannelProvider {
   /** True if the TransportProvider has no endpoint set. */
   boolean needsEndpoint();
 
+  /** True if the TransportProvider needs an {@link EndpointContext} */
+  boolean needsEndpointContext();
+
```

Comment:
Is the reason this was kept in because FixedTransportChannelProvider, LocalChannelProvider, and FakeTransportChannelProvider don't need the EndpointContext.

I think we can remove this method. I think FixedTransportChannelProvider should no-op. LocalChannelProvider and FakeTransportChannelProvider should use the EndpointContext. I'm guessing some tests may fail with this change, but probably should get resolved with just setting a dummy EndpointContext.

Reply:
Is the reason this was kept in because FixedTransportChannelProvider, LocalChannelProvider, and FakeTransportChannelProvider don't need the EndpointContext.
> mainly yes, but I've removed the method and adjusted the implementers of the interface and some failing tests as a result.

Done in https://github.com/googleapis/sdk-platform-java/pull/3326/commits/393190a1ad03838ff5a3e13222c75e349f0be72e